midend/local_copyprop.cpp

```diff
@@ -204,8 +210,34 @@ IR::AssignmentStatement *DoLocalCopyPropagation::preorder(IR::AssignmentStatemen
     return postorder(as);
 }
 
+/* Function to check if left and right side of
+ * assignment statement are equivalent.
+ * FIXME -- This function is temporary, proper deep comparison may be required */
+bool DoLocalCopyPropagation::equiv(const IR::Expression *left, const IR::Expression *right) {
+    // Compare names of variables (at this pass, all names are unique)
+    auto l1 = left->to<IR::PathExpression>();
+    auto r1 = right->to<IR::PathExpression>();
+    if (l1 && r1) {
+        return l1->path->name == r1->path->name;
+    }
+    // Compare references to arrays with indices
+    auto l2 = left->to<IR::Operation_Binary>();
+    auto r2 = right->to<IR::Operation_Binary>();
+    if (l2 && r2) {
+        return equiv(l2->left, r2->left) && equiv(l2->right, r2->right);
```

Comment:
I think this is safe solely because the only `Operation_Binary` that could appear here (lhs of assignment) is `IR::ArrayIndex`.  For generality, should add `&& typeid(*l2) == typeid(*r2)`